src/react/components/pages/editorPage/editorFooter.tsx

```diff
@@ -84,22 +76,43 @@ export default class EditorFooter extends React.Component<IEditorFooterProps, IE
         );
     }
 
-    private onShiftTagClicked(tag: ITag) {
+    private onShiftTagClicked = (tag: ITag) => {
         this.setState({
             selectedTag: tag,
         }, () => {
             this.tagEditorModal.current.open(tag);
         });
     }
 
-    private onTagModalOk(oldTag: ITag, newTag: ITag) {
-        this.editorTagsInput.current.updateTag(oldTag, newTag);
+    private onTagModalOk = (oldTag: ITag, newTag: ITag) => {
+        this.tagsInput.current.updateTag(oldTag, newTag);
         this.tagEditorModal.current.close();
     }
 
-    private onTagsChanged(tags) {
+    private onTagsChanged = (tags) => {
         this.setState({
             tags,
         }, () => this.props.onTagsChanged(this.state));
     }
+
+    /**
+     * Shows the of the tag in the span of the first 10 tags
+     * @param name Name of tag
+     * @param index Index of tag
+     */
+    private getTagSpan = (name: string, index: number) => {
+        let className = "tag-span";
+        let displayName = name;
+        if (index < 10) {
+            const displayIndex = (index === 9) ? 0 : index + 1;
```

Comment:
what is the reasoning behind this index math?

Reply:
The display index is not the same as the index in the array. We want the 10th tag to be 0 and every other to be `index + 1`